### PR TITLE
webmin - use the new reinstall-functions and smaller fixes

### PR DESCRIPTION
**Pull request by szaimen**

### CODE REVIEW

**Comment by szaimen** [comment]
Should the `/etc/apt/sources.list.d/webmin.list` file also get removed upon reinstalling/uninstalling?
https://github.com/nextcloud/vm/pull/1457/files#diff-470c274da614a3e50289211998c445f1R52

File: apps/webmin.sh [hunk 1]
```diff
@@ -5,6 +5,10 @@
 # shellcheck disable=2034,2059
 true
 SCRIPT_NAME="Webmin"
+SCRIPT_EXPLAINER="Webmin is a web-based interface for system administration for Unix.
+Using any modern web browser, you can setup user accounts, Apache, DNS, file sharing and much more.
+Webmin removes the need to manually edit Unix configuration files like /etc/passwd, and lets you manage a system from the console or remotely.
+See the following page with standard modules for a list of all the functions built into Webmin: https://webmin.com/standard.html"
```
Comment (szaimen) [inline apps/webmin.sh:11]: This description is from the official webmin website. Do you think that we can use that?
https://webmin.com/ (section What is Webmin?)

Comment (enoch85) [reply apps/webmin.sh:11]: LGTM

File: lib.sh [hunk 2]
```diff
@@ -313,15 +313,19 @@ explainer_popup() {
 }
 
 reinstall_remove_menu() {
-    choice=$(whiptail --title "$TITLE" --menu "It seems like '$SCRIPT_NAME' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like $SCRIPT_NAME is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
     "Reinstall $SCRIPT_NAME" "" \
     "Uninstall $SCRIPT_NAME" "" 3>&1 1>&2 2>&3)
+    if [ -z "$choice" ]
```
Comment (enoch85) [inline lib.sh:319]: We should add this everywhere. Isn't this exiting when you hit cancel? Currently that's not working.

File: apps/adminer.sh [hunk 3]
```diff
@@ -32,7 +32,7 @@ then
     a2disconf adminer.conf
     rm -f $ADMINER_CONF
     rm -rf $ADMINERDIR
-    check_command apt-get purge adminer -y
+    check_command apt --purge autoremove -y adminer
```
Comment (enoch85) [inline apps/adminer.sh:35]: No, we use apt-get for purging.

Please revert. Thanks!

**Comment by enoch85** [comment]
> Should the /etc/apt/sources.list.d/webmin.list file also get removed upon reinstalling/uninstalling?

Yes please :+1:

File: lib.sh (hunk 2 above)
Comment (szaimen) [reply lib.sh:319]: done

File: apps/adminer.sh (hunk 3 above)
Comment (szaimen) [reply apps/adminer.sh]: done

**Comment by szaimen** [comment]
> > Should the /etc/apt/sources.list.d/webmin.list file also get removed upon reinstalling/uninstalling?
> 
> Yes please 👍

done

**Comment by szaimen** [comment]
What about this line: how to remove this?
`curl -fsSL http://www.webmin.com/jcameron-key.asc | sudo apt-key add -`

**Comment by enoch85** [comment]
> curl -fsSL http://www.webmin.com/jcameron-key.asc | sudo apt-key add -

You can list all keys with `apt-key list`,  with that you could probably `grep` for 'cameron' or similar and then do a `apt-key del`. This is from the top of my head, would need confirmation.

**Comment by enoch85** [comment]
> curl -fsSL http://www.webmin.com/jcameron-key.asc | sudo apt-key add -

I will release 19.0.3 today, so let's figure that out in a separate PR.

**Pull request merged**